### ISSUE TITLE
Investigate these surprising results

### ISSUE THREAD

**Issue by Entomy** [Entomy/Stringier#31]
![image](https://user-images.githubusercontent.com/25399505/71372133-7b6d7000-2581-11ea-9c3f-e61a7109e8db.png)

In every single case, the `Fuzzer` was faster than the literal match. This is bizarre and unexpected. It was my understanding that the Hamming edit-distance algorithm, which the `Fuzzer` fundamentally uses, is more expensive than the approach the CoreCLR uses. This should make sense as all edit-distance algorithms have to not just compare the characters, but also accumulate the edit count.

It may only be the case that this effects string parsing. But it should still be investigated as there's definate and obvious benefits.

**Comment by Entomy** [comment]
I've confirmed essentially identical results across several machines, so it's not a machine specific behavior. Some of these weren't even the same CPU architecture.

**Comment by Entomy** [comment]
![Annotation 2020-01-02 091958](https://user-images.githubusercontent.com/25399505/71671444-0fd59080-2d41-11ea-9208-49862b7a2232.png)

So, I don't think it's the result of the Hamming edit-distance per-say. It's way too varied, compared to the consistent results that were shown above.

**Comment by Entomy** [comment]
I swapped out the implementation of `Equals()` that is happening for all `String` and `StringLiteral` `Consume()` calls. It's definitely not the result of a (Modified-)Hamming edit-distance algorithm.

**Comment by Entomy** [comment]
I'm considering this closed. After a bunch of testing it seems like I've found the biggest culprits. Method calls obviously have a performance impact, as a stack frame needs to be set up, as well as a jump and eventual return. Too many of these wastes CPU cycles, and is where inlining can be beneficial. However excessive inlining can also hurt performance, as methods should be a suitable size for fitting into CPU caches, and misalignment or cross-page issues become more likely as method sizes increase. I believe I've found the best methods to aggressively inline without causing problems on most machines. This is something that should be cautiously watched in the future, however.
